core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -189,9 +186,9 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
             )
         }
       billingProfileId <- billingRepository.getBillingProfileId(projectName)
-      (deleteType, projectLifecycle) = billingProfileId match {
-        case None    => (GoogleBillingProjectDelete, googleBillingProjectLifecycle)
-        case Some(_) => (BpmBillingProjectDelete, bpmBillingProjectLifecycle)
+      projectLifecycle = billingProfileId match {
```

Comment:
I realized I could make the behavior more consistent by using the return value of `BillingProjectLifecycle.initiateDelete` to indicate whether or not the deletion has to wait on a job in order to finish.
So if we get `Some(jobId)`: schedule the job. If we get `None`: finalize the delete immediately.

This eliminates the need for a dummy job id, and makes the logic simpler.